tests/Agent/IntegrationTests/IntegrationTests/HttpClientInstrumentation/HttpClientInstrumentation.cs

```diff
@@ -0,0 +1,173 @@
+                () => Assert.True(externalSpanEvents[0].IntrinsicAttributes.TryGetValue("component", out var value) && value.ToString().StartsWith("System.Net.Http.")),
+                () => Assert.All(externalSpanEvents, AssertSpanEventsContainHttpStatusCodeForCompletedRequests)
+            );
+
+            var agentWrapperErrorRegex = AgentLogBase.ErrorLogLinePrefixRegex + @"An exception occurred in a wrapper: (.*)";
+            var wrapperError = _fixture.AgentLog.TryGetLogLine(agentWrapperErrorRegex);
+
+            Assert.Null(wrapperError);
+
+            void AssertSpanEventsContainHttpStatusCodeForCompletedRequests(SpanEvent spanEvent)
+            {
+                var url = (string)spanEvent.AgentAttributes["http.url"];
+                if (url.Contains("bing"))
+                {
+                    Assert.DoesNotContain("http.statusCode", spanEvent.AgentAttributes.Keys);
+                }
+                else
+                {
+                    Assert.Contains("http.statusCode", spanEvent.AgentAttributes.Keys);
+                }
+            }
+        }
+    }
+
+    [NetCoreTest]
+    public class HttpClientInstrumentationTests_Net31 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureCore31>
+    {
+        public HttpClientInstrumentationTests_Net31(ConsoleDynamicMethodFixtureCore31 fixture, ITestOutputHelper output)
+            : base(fixture, output)
+        {
+        }
+    }
+
+    [NetCoreTest]
+    public class HttpClientInstrumentationTests_Net50 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureCore50>
+    {
+        public HttpClientInstrumentationTests_Net50(ConsoleDynamicMethodFixtureCore50 fixture, ITestOutputHelper output)
+            : base(fixture, output)
+        {
+        }
+    }
+
+    [NetCoreTest]
+    public class HttpClientInstrumentationTests_NetCoreLatest : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureCoreLatest>
+    {
+        public HttpClientInstrumentationTests_NetCoreLatest(ConsoleDynamicMethodFixtureCoreLatest fixture, ITestOutputHelper output)
+            : base(fixture, output)
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class HttpClientInstrumentationTests_FW462 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureFW462>
+    {
+        public HttpClientInstrumentationTests_FW462(ConsoleDynamicMethodFixtureFW462 fixture, ITestOutputHelper output)
+            : base(fixture, output)
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class HttpClientInstrumentationTests_FW471 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureFW471>
+    {
+        public HttpClientInstrumentationTests_FW471(ConsoleDynamicMethodFixtureFW471 fixture, ITestOutputHelper output)
+            : base(fixture, output)
+        {
+        }
+    }
+
+    [NetFrameworkTest]
+    public class HttpClientInstrumentationTests_FW48 : HttpClientInstrumentationTestsBase<ConsoleDynamicMethodFixtureFW48>
```

Comment:
Should this be `_FWLatest` to match the pattern?

Reply:
As there are no new releases of .NET Framework planned, I think this is fine as is